lib/tx_build/claim_predicate.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Stellar.TxBuild.ClaimPredicate do
+  @moduledoc """
+  `ClaimPredicate` struct definition.
+  """
+  alias Stellar.TxBuild.{ClaimPredicates, OptionalClaimPredicate}
+  alias StellarBase.XDR.{ClaimPredicate, ClaimPredicateType, Int64, Void}
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type validation :: {:ok, any()} | {:error, any()}
+  @type predicate :: :unconditional | :conditional
+  @type type :: :time | :and | :or | :not
+  @type time_type :: :relative | :absolute
+  @type value :: t() | list(t()) | pos_integer()
+
+  @type t :: %__MODULE__{
+          predicate: predicate(),
+          value: value(),
+          type: type(),
+          time_type: time_type()
+        }
+
+  defstruct [:predicate, :value, :type, :time_type]
+
+  @impl true
+  def new(args, opts \\ [])
+
+  def new(:unconditional, _opts), do: %__MODULE__{predicate: :unconditional}
+
+  def new({value, type, time_type}, _opts),
+    do: new(value: value, type: type, time_type: time_type)
+
+  def new([value: value, type: type], _opts)
+      when type === :and or type === :or or type === :not do
```

Comment:
@einerzg you can simplify this guard condition using something like `type in ~w(and or not)a` or even simpler `type != :time`

Reply:
I suggest using the first proposal `type in ~w(and or not)a` because the meaning is easier to understand.